packages/vite/index.cjs

```diff
@@ -17,12 +20,42 @@ const asyncFunctions = [
   'formatPostcssSourceMap',
   'loadConfigFromFile',
   'preprocessCSS',
+  'createBuilder',
 ]
 asyncFunctions.forEach((name) => {
   module.exports[name] = (...args) =>
     import('./dist/node/index.js').then((i) => i[name](...args))
 })
 
+// variables and sync functions that cannot be used from cjs build
+const disallowedVariables = [
+  // was not exposed in cjs from the beginning
+  'parseAst',
+  'buildErrorMessage',
+  'sortUserPlugins',
+  // Environment API related variables that are too big to include in the cjs build
+  'DevEnvironment',
+  'BuildEnvironment',
+  'createIdResolver',
+  'createRunnableDevEnvironment',
+  // can be redirected from ESM, but doesn't make sense as it's Environment API related
+  'fetchModule',
+  'moduleRunnerTransform',
+  // can be exposed, but doesn't make sense as it's Environment API related
+  'createServerHotChannel',
+  'createServerModuleRunner',
+  'isRunnableDevEnvironment',
+]
+disallowedVariables.forEach((name) => {
+  Object.defineProperty(module.exports, name, {
+    get() {
+      throw new Error(
+        `${name} is not available in the CJS build. ` + description,
```

Comment:
```suggestion
        `${name} is not available in the CJS build of Vite.` + description,
```

The description already has a leading space, so I think we can remove it here. I also added a "of Vite" here so it's easier to see where the error came from.

Reply:
Sounds good 👍